src/QueryCache.php

```diff
@@ -37,127 +44,101 @@ function activate(AbstractBackend $backend)
             $this->backend = $backend;
         }
 
-        add_action(
-            'do_graphql_request',
-            [$this, '__action_do_graphql_request'],
-            10,
-            4
-        );
-
-        add_action(
-            'graphql_process_http_request_response',
-            [$this, '__action_graphql_process_http_request_response'],
-            // Use large value as this should be the last response filter
-            // because we want to save the last version of the response to the
-            // cache.
-            1000,
-            5
-        );
+        add_filter('process_graphql_execute_request', [$this, '__filter_do_graphql_request'], 10, 2);
 
         add_action('graphql_response_set_headers', [
             $this,
             '__action_graphql_response_set_headers',
         ]);
     }
 
-    function __action_do_graphql_request(
-        ?string $query,
-        $operation,
-        $variables,
-        $params
-    ) {
+    /**
+    * @param ExecutionResult|null $response
+    * @param OperationParams $params
+    * @return ExecutionResult|array|null
+    */
+    function __filter_do_graphql_request($response, $params)
+    {
+
+        $query = $params->query;
+
         if (empty($query)) {
-            return;
+            return $response;
         }
 
-        $user_id = $this->per_user ? get_current_user_id() : $this->user_id_for_shared_cache;
```

Comment:
The default value of `$user_id_for_shared_cache` is `0`, but it could be overridden. So, the standard behaviour is to use a user ID of `0` when the cache is not user-specific. The name of this variable isn't brilliant, admittedly.